pages/index.js

```diff
@@ -2,16 +2,56 @@ import useSWR from "swr";
 import Error from "@/components/Error";
 import Loading from "@/components/Loading";
 import EventList from "@/components/EventList";
+import FilterBar from "@/components/FilterBar";
+import { useState } from "react";
 
 export default function HomePage() {
-  const { data: events, isLoading, error } = useSWR(`/api/events`);
+  const [filter, setFilter] = useState({});
+
+  const {
+    data: events,
+    isLoading,
+    error,
+  } = useSWR(
+    `/api/events?queryCategory=${
+      filter?.category ? filter.category : ""
+    }&queryCity=${filter?.city ? filter.city : ""}`
+  );
```

Comment:
Too much ternary = nobueno

Abstract it in a function like:

```js
function getURL(filter = {}) {
  const params = new URLSearchParams();
  filter.category && params.append('category', filter.category);
  filter.city && params.append('city', filter.city);
  return `/api/events${params.size ? `?${params}` : ''}`;
}

console.log(getURL({ category: 'a' }));             // /api/events?category=a
console.log(getURL({ category: 'a', city: 'b' }));  // /api/events?category=a&city=b
console.log(getURL({ city: 'b' }));                 // /api/events?city=b
console.log(getURL());                              // /api/events
```

[See URLSearchParams docs](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) for more ^^